x-pack/legacy/plugins/ml/public/application/datavisualizer/index_based/components/fields_panel/fields_panel.tsx

```diff
@@ -78,51 +79,51 @@ export const FieldsPanel: FC<Props> = ({
   }
 
   return (
-    <EuiPanel data-test-subj={`mlDataVisualizerFieldsPanel ${fieldTypes}`}>
-      <EuiTitle>
-        <h2>{title}</h2>
-      </EuiTitle>
-      <EuiSpacer size="s" />
-      <EuiFlexGroup alignItems="center">
-        <EuiFlexItem grow={false}>
-          <span>
-            {showAllFields === true ? (
-              <FormattedMessage
-                id="xpack.ml.datavisualizer.fieldsPanel.showAllCountDescription"
-                defaultMessage="{wrappedCardsCount} {cardsCount, plural, one {field} other {fields}} ({wrappedPopulatedFieldCount} {populatedFieldCount, plural, one {exists} other {exist}} in documents)"
-                values={{
-                  cardsCount: fieldVisConfigs.length,
-                  wrappedCardsCount: <b>{fieldVisConfigs.length}</b>,
-                  populatedFieldCount,
-                  wrappedPopulatedFieldCount: <b>{populatedFieldCount}</b>,
-                }}
-              />
-            ) : (
-              <FormattedMessage
-                id="xpack.ml.datavisualizer.fieldsPanel.fieldsCountDescription"
-                defaultMessage="{wrappedCardsCount} {cardsCount, plural, one {field exists} other {fields exist}} in documents ({wrappedTotalFieldCount} in total)"
-                values={{
-                  cardsCount: fieldVisConfigs.length,
-                  wrappedCardsCount: <b>{fieldVisConfigs.length}</b>,
-                  wrappedTotalFieldCount: <b>{totalFieldCount}</b>,
-                }}
-              />
+    <div data-test-subj={`mlDataVisualizerFieldsPanel ${fieldTypes}`}>
+      <EuiFlexGroup alignItems="center" justifyContent="spaceBetween">
+        <EuiFlexItem>
+          <EuiFlexGroup alignItems="center" gutterSize="m">
+            <EuiFlexItem grow={false}>
+              <EuiTitle size="m">
+                <h2>{title}</h2>
+              </EuiTitle>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiBadge>
+                <b>{fieldVisConfigs.length}</b>
+              </EuiBadge>
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <EuiText size="s" color="subdued">
+                <FormattedMessage
+                  id="xpack.ml.datavisualizer.fieldsPanel.totalFieldLabel"
+                  defaultMessage="Total fields: {wrappedTotalFields}"
+                  values={{
+                    wrappedTotalFields: <b>{populatedFieldCount}</b>,
+                  }}
+                />
+              </EuiText>
+            </EuiFlexItem>
+            {populatedFieldCount < totalFieldCount && (
+              <EuiFlexItem grow={false}>
+                <EuiSwitch
+                  id={`${title}_show_empty_fields`}
+                  label={i18n.translate(
+                    'xpack.ml.datavisualizer.fieldsPanel.showEmptyFieldsLabel',
+                    {
+                      defaultMessage: 'Show empty fields',
```

Comment:
When `Show empty fields` is enabled, the information is not correct.

![image](https://user-images.githubusercontent.com/7405507/72253593-d2eb8280-35f9-11ea-89f9-935017b42efa.png)

compared to:

![image](https://user-images.githubusercontent.com/7405507/72253641-eac30680-35f9-11ea-853c-b960aa6f8d56.png)

Here, the grey box should show '7', with a tooltip saying something like `7 fields exists in documents sampled` , whilst total fields should show '8'.

Reply:
Fixed in https://github.com/elastic/kibana/pull/54358/commits/5a39b629f5283016edd7dbfeaadd57ba09dc54f8

Reply:
As above, these numbers are the wrong way round.

Reply:
https://github.com/elastic/kibana/pull/54358/commits/c16861c376ac8baff8989dcb59522ac8f09680a0